File: 5. Right Triangle Patterns (Bottom-Left)/43. Right Triangle BL Increasing Numbers/blin.java
import java.util.*;
public class blin{
    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a Number : ");
        int n = sc.nextInt();

        for(int i=1; i<=n; i++){
            for(int j=i; j<=n; j++){ // for(int j=1; j<=(n-i+1); j++)
                System.out.print(j-i+1 + " ");
            }
            System.out.println();
        }

        sc.close();
    }
}

/*

Use (i+j<=n+1) for optimized code & for n matrix 

Output

1 2 3 4 5 
1 2 3 4 
1 2 3 
1 2 
1  

*/

File: 5. Right Triangle Patterns (Bottom-Left)/43. Right Triangle BL Increasing Numbers/blin.py
n = int(input("Enter a Number : "))

for i in range(1,n+1):
    for j in range(i,n+1): # for j in range(1,n-i+2):
        print(j-i+1,end=" ")
    print()

'''

Use (i+j<=n+1) for optimized code & for n matrix 

Output:

1 2 3 4 5 
1 2 3 4 
1 2 3 
1 2 
1 
 
'''